forum/Sources/Tips.php

```diff
@@ -0,0 +1,81 @@
+<?php 
+
+if (!defined('SMF'))
+	die('Hacking attempt...');
+
+function Tips()
+{
+	global $context, $smcFunc, $user_info, $txt;
+
+	$context['page_title'] = $txt['tip_list_title'];
+	
+	loadPosts();
+
+	loadTemplate('Tips');
+}
+
+function loadPosts()
+{
+	global $memberContext, $context, $smcFunc, $user_info;
+	
+	$tippedPosts = array();
+	
+	// Find all tips in descending order
+	// TO-DO: custom-limit: pagination support, etc.
+	// Get most recent tipped posts
+	$query = 'SELECT id_message_tip, id_message, id_member, coins FROM {db_prefix}message_tips';
+
+	// Searching for posts tipped by a specific user?
+	if (!empty($_GET['tipper']) && !$context['user']['is_guest'])
+	{
+		// Find user id for searched user
+		$userSearch = $smcFunc['db_query']('', 'SELECT id_member from smf_members where real_name = \'' . mysql_escape_string(urldecode($_GET['tipper'])) . '\'');

```

Comment:
Why use single quotes and escape them instead of using double quotes?

Reply:
I've used single quotes and escaped because SMF itself does that literally everywhere.

I thought it best to maintain that consistency, but I guess it's not necessary.